src/vivarium_census_prl_synth_pop/components/household.py

```diff
@@ -0,0 +1,148 @@
+            pd.Series(data_values.GQ_HOUSING_TYPE_MAP.keys())
+            .astype(HOUSEHOLD_ID_DTYPE)
+            .rename("household_id")
+        )
+        self._households = pd.DataFrame(
+            {
+                "address_id": data_values.GQ_HOUSING_TYPE_MAP.keys(),
+                "housing_type": data_values.GQ_HOUSING_TYPE_MAP.values(),
+            },
+            index=gq_household_ids,
+        ).astype(HOUSEHOLD_DETAILS_DTYPES)
+
+        self.household_details = builder.value.register_value_producer(
+            "household_details",
+            source=self.get_household_details,
+            requires_columns=["household_id", "tracked"],
+        )
+
+    ##################################
+    # Pipeline sources and modifiers #
+    ##################################
+
+    def get_household_details(self, idx: pd.Index) -> pd.DataFrame:
+        """Source of the household_details pipeline"""
+        pop = self.population_view.subview(["tracked", "household_id"]).get(idx)
+        household_details = (
+            pop[["household_id"]]
+            .astype(HOUSEHOLD_ID_DTYPE)
+            .join(
+                self._households,
+                on="household_id",
+            )
+        )
+        household_details = household_details.astype(HOUSEHOLD_DETAILS_DTYPES)
+        return household_details
+
+    ##################
+    # Public methods #
+    ##################
+
+    def create_households(
+        self, num_households: int, housing_type: str = "Standard"
+    ) -> pd.Series:
+        """
+        Create a specified number of new households, with new, unique address_ids.
+
+        Parameters
+        ----------
+        num_households
+            The number of households to create.
+        housing_type
+            The housing type of the new households.
+
+        Returns
+        -------
+        household_ids for the new households.
+        The length of this Series will be num_households.
+        """
+        household_ids = self._next_available_ids(num_households, taken=self._households.index)
+        address_ids = self._next_available_ids(
+            num_households, taken=self._households["address_id"]
+        )
+
+        new_households = pd.DataFrame(
+            {
+                "address_id": address_ids,
+                "housing_type": housing_type,
+            },
+            index=household_ids.astype(HOUSEHOLD_ID_DTYPE),
+        ).astype(HOUSEHOLD_DETAILS_DTYPES)
+
+        self._households = pd.concat([self._households, new_households])
+
+        return household_ids
+
+    def update_household_addresses(self, household_ids: pd.Series) -> None:
+        """
+        Changes the address_id associated with each household_id in household_ids
+        to a new, unique value.
+
+        Parameters
+        ----------
+        household_ids
+            The IDs of the households that should move to new addresses.
+        """
+        self._households, _ = update_address_ids(
```

Comment:
I'm a bit conflicted about the utility of this function. It's used twice - here and in businesses, but there we need the max id and here we don't (and the function itself is quite simple). Would it be better just have the code duplicated rather than calling out to a function in utilities?

Reply:
Great point. I think I can write this method more simply/clearly without using the utilities function. That will leave just one use of the utilities function, so I can move it into the businesses component as a helper.